bugalink-backend/trips/views.py

```diff
@@ -88,79 +94,124 @@ def get(self, request, *args, **kwargs):
+                payment_intent = stripe.PaymentIntent.create(
+                    payment_method=payment_method.id,
                     amount=amount,
-                    currency="eur",
-                    source=request.data.get(
-                        "stripe_token"
-                    ),  # TODO replace with a valid card token
-                    description="Viaje Bugalink - " + str(trip.departure_datetime),
+                    currency="usd",
+                    confirmation_method="manual",
+                    confirm=True
                 )
 
-            except stripe.error.CardError:
+                # Check if the payment is succeeded
+                if payment_intent.status == "succeeded":
+                    print("Payment succeeded!")
+                else:
+                    return Response(
+                        status=status.HTTP_400_BAD_REQUEST,
+                        data={"error": "Método de pago fallido"},
+                    )
+
+            except stripe.error.StripeError as e:
                 return Response(
                     status=status.HTTP_400_BAD_REQUEST,
-                    data={"error": "Tarjeta inválida"},
+                    data={"error": "Stripe error"},
                 )
 
         def pay_with_paypal(price):
-            paypal_client_id = "your_paypal_client_id"  # TODO change it
-            paypal_secret_key = "your_paypal_secret_key"  # TODO change it
-
-            environment = paypal.Environment(
-                client_id=paypal_client_id, client_secret=paypal_secret_key
-            )
-            client = paypal.PayPalHttpClient(environment)
-
-            # Get the amount to charge
-            amount = price
-
-            # Create a PayPal order
-            order = paypal.OrderRequest(
-                {
-                    "intent": "CAPTURE",
-                    "purchase_units": [
-                        {"amount": {"currency_code": "EUR", "value": str(amount)}}
-                    ],
-                }
-            )
-            response = client.execute(paypal.OrdersCreateRequest(order))
-            order_id = response.result.id
+            paypal_client_id = "AdWSL48duytv4qy76be71a2S3Tt5nTYn-1gGv-53vL_dxNWYzZpAGrUZYrZBvGjkNwOSxJE1s_RSCkL8"
+            paypal_secret_key= "EHps0LO5OsQsUOrDTu9J6BY_mD0OkcF9aNzOT7rkRtDYKCOxoiqCUXsnz-nkhZX5rhlA741NosbaxBpb" 
+
+            # Set up PayPal API credentials
+            paypalrestsdk.configure({
+                "mode": "sandbox",  
+                "client_id": paypal_client_id,
+                "client_secret": paypal_secret_key,
+            })
+
+            # Create a payment object
+            payment = Payment({
+                "intent": "sale",
+                "payer": {
+                    "payment_method": "paypal",
+                },
+                "redirect_urls": {
+                    "return_url": "http://app.bugalink.es",
```

Comment:
What about development?? This has to use an environment variable defining that the front URL is either this or `localhost:3000`. Also, you are missing the "s" in "https://"